test/aria/templates/TplClassGeneratorParseOnlyTest.js

```diff
@@ -0,0 +1,101 @@
+
+            var mainMacroStatement = result.tree.properties.macros.main.definition;
+            this.assertEquals(mainMacroStatement.name, "macro");
+            this.assertEquals(mainMacroStatement.paramBlock, "main()");
+            this.assertEquals(result.tree.source.substring(mainMacroStatement.firstCharParamIndex, mainMacroStatement.lastCharParamIndex), "main()");
+            this.assertEquals(mainMacroStatement.properties.name, "main");
+            this.assertJsonEquals(mainMacroStatement.properties.args, []);
+
+            var callTestStatement = mainMacroStatement.content[0];
+            this.assertEquals(callTestStatement.name, "call");
+            this.assertEquals(callTestStatement.properties.name, "test");
+            this.assertEquals(callTestStatement.properties.container, undefined);
+            this.assertEquals(callTestStatement.properties.args, 'null, "value", "data"');
+
+            var callMainStatement = mainMacroStatement.content[1];
+            this.assertEquals(callMainStatement.name, "call");
+            this.assertEquals(callMainStatement.properties.name, "main");
+            this.assertEquals(callMainStatement.properties.container, "$Y");
+            this.assertEquals(callMainStatement.properties.args, "");
+
+            var testMacroStatement = result.tree.properties.macros.test.definition;
+            this.assertEquals(testMacroStatement.name, "macro");
+            this.assertEquals(testMacroStatement.paramBlock, "test(  a ,   bb , c   )");
+            this.assertEquals(testMacroStatement.properties.name, "test");
+            this.assertJsonEquals(testMacroStatement.properties.args, ["a", "bb", "c"]);
+
+            var textStatement = testMacroStatement.content[0];
+            this.assertEquals(textStatement.name, "#TEXT#");
+            this.assertEquals(textStatement.paramBlock, "Date:");
+
+            var exprStatement = testMacroStatement.content[1];
+            this.assertEquals(exprStatement.name, "#EXPRESSION#");
+            this.assertEquals(exprStatement.paramBlock, 'a|empty:new Date()|dateformat:"dd/MM/yyyy"');
+            this.assertEquals(exprStatement.properties.expression, 'a');
+            this.assertJsonEquals(exprStatement.properties.modifiers, [{
+                "name": "empty",
+                "args": 'new Date()'
+            }, {
+                "name": "dateformat",
+                "args": '"dd/MM/yyyy"'
+            }]);
+
+            var widgetStatement = testMacroStatement.content[2];
+            this.assertEquals(widgetStatement.name, "@myLib:MyWidget");
+            this.assertEquals(widgetStatement.properties.libName, "myLib");
+            this.assertEquals(widgetStatement.properties.libClasspath, "x.y.MyWidgetsLibThatDoesNotExistButShouldNotBeChecked");
+            this.assertEquals(widgetStatement.properties.widgetName, "MyWidget");
+
+            this.assertTruthy(result, "The callback should have been called synchronously.");
```

Comment:
Shouldn't you test this first? I (only) now understand why you were assuming the `result` object was immediately available.